src/plugins/intel_gna/tests/functional/subgraph_tests/add_transpose_detection.cpp

```diff
@@ -0,0 +1,231 @@
+        auto input = std::make_shared<Parameter>(ng_precision, ov::Shape{input_shape});
+
+        const auto weights_size = ov::shape_size(filter_size) * out_channels_num * input_shape[c_index_in_nchw];
+        auto weights_values = CommonTestUtils::generate_float_numbers(weights_size, -0.2f, 0.2f);
+
+        auto convolution = ngraph::builder::makeConvolution(input,
+                                                            ng_precision,
+                                                            filter_size,
+                                                            strides,
+                                                            pads_begin,
+                                                            pads_end,
+                                                            dilations,
+                                                            pad_type,
+                                                            out_channels_num,
+                                                            false,
+                                                            weights_values);
+        auto add = std::make_shared<Add>(input, convolution);
+        auto res = std::make_shared<Result>(add);
+        function = std::make_shared<ov::Model>(ov::ResultVector{res}, ov::ParameterVector{input});
+    }
+};
+
+class ConvInputAddTransposing : public InputConvAddTransposing {
+protected:
+    void SetUp() override {
+        InferenceEngine::Precision precision;
+        std::vector<size_t> input_shape;
+        std::tie(precision, targetDevice, configuration, input_shape) = this->GetParam();
+
+        auto ng_precision = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(precision);
+        auto input = std::make_shared<Parameter>(ng_precision, ov::Shape{input_shape});
+        auto const_node = std::make_shared<Constant>(ng_precision, ov::Shape{input_shape});
+        const auto weights_size = ov::shape_size(filter_size) * out_channels_num * input_shape[c_index_in_nchw];
+        auto weights_values = CommonTestUtils::generate_float_numbers(weights_size, -0.2f, 0.2f);
+
+        auto convolution = ngraph::builder::makeConvolution(input,
+                                                            ng_precision,
+                                                            filter_size,
+                                                            strides,
+                                                            pads_begin,
+                                                            pads_end,
+                                                            dilations,
+                                                            pad_type,
+                                                            out_channels_num,
+                                                            false,
+                                                            weights_values);
+        auto add = std::make_shared<Add>(convolution, input);
```

Comment:
Maybe just parametrize it using 2 bool parameters (input/const, first/second) to avoid copy paste?

Reply:
optimized )